src/Elastic.Apm/Report/PayloadSenderV2.cs

```diff
@@ -148,7 +148,7 @@ IApmLogger logger
 
 		private bool _getApmServerVersion;
 		private bool _getCloudMetadata;
-		private bool _allowFilterAdd;
+		private bool _allowFilterAdd = true;
```

Comment:
The intention was more to allow filters to be added until the first time we start sending data out. 

Since `Agent.AddFilter()` can be called prior and/or after `Agent.Setup()` this is a lenient solution. Ideally we'd only allow filters in the payload constructor but our public API does not allow us to enforce that without breaking changes.

It was also added to ensure we do not modify the `List<Func<>>` while we are iterating over it in `WorkLoopIteration`.

Reply:
Oh, I think I get it.

> Ideally we'd only allow filters in the payload constructor 

Yeah, that'd be much cleaner - I feel for a user it's hard to know when the agent start sending out data. You can initalize the agent, but if you are fast enough you can still add filters :)

In any case, I understand the historical baggage here. Thanks for clarifying.